resolwe_bio/management/commands/generate_de.py

```diff
@@ -0,0 +1,157 @@
+
+    @staticmethod
+    def get_random_word(length):
+        return ''.join(random.choice(string.lowercase) for _ in range(length))
+
+    @staticmethod
+    def generate_expressions(gene_ids, path):
+        """Generate random expression data"""
+        genes = {}
+        with gzip.open(os.path.join(path, 'expressions.tab.gz'), 'wb') as f:
+            f.write('{}\t{}\n'.format('Gene', 'Expression'))
+            with gzopen(gene_ids) as gene_ids:
+                all_genes = [line.strip() for line in gene_ids]
+                for gene in all_genes:
+                    expression = random.gammavariate(1, 100)
+                    f.write('{}\t{}\n'.format(gene, expression))
+                    genes[escape_mongokey(gene)] = expression
+
+        with open(os.path.join(path, 'expressions.json'), 'w') as json_file:
+            js_out = '{{"exp_json":{}}}'.format(json.dumps({'genes': genes}, separators=(',', ':')))
+            json_file.write(js_out)
+
+    def set_user(self):
+        users = User.objects.filter(is_superuser=True).order_by('date_joined')
+        if not users.exists():
+            self.stderr.write("Admin does not exist: create a superuser")
+            exit(1)
+        return users.first()
+
+    def set_name(self, de_name):
+        return 'DE_{}_{}'.format(self.get_random_word(4), de_name)
+
+    def create_expressions(self, n, data_dir, human_genes):
+        expressions = []
+        sample_name = self.get_random_word(4)
+
+        for i in range(n):
+            # Create expressios
+            exp = Data.objects.create(
+                name='Smpl_Ex_{}_rep{}'.format(sample_name, i+1),
+                process=Process.objects.get(slug='upload-expression'),
+                contributor=self.set_user(),
+                status=Data.STATUS_PROCESSING,
+                input={'exp': {'file': 'expressions.tab.gz'}, 'exp_type': 'FPKM'})
+
+            os.mkdir(os.path.join(data_dir, str(exp.id)))
+            self.generate_expressions(human_genes, os.path.join(data_dir, str(exp.id)))
+
+            json_object = Storage.objects.create(
+                json=json.load(open(os.path.join(data_dir, str(exp.id), 'expressions.json'))),
+                contributor=self.set_user(),
+                data=exp)
+
+            exp.output = {
+                'exp': {'file': 'expressions.tab.gz'},
+                'exp_type': 'FPKM',
+                'exp_json': json_object.id
+            }
+            exp.status = Data.STATUS_DONE
+            exp.save()
+
+            sample = Sample.objects.filter(data=exp)[0]
+            sample.presample = False
+            sample.save()
+
+            with open(os.path.join(data_dir, str(exp.id), 'stdout.txt'), 'w') as stdout:
+                stdout.write('Upload gene expressions. Sample was created '
+                             'with the generate_samples django-admin command.')
+
+            print "Created sample: {} (id={})".format(sample.name, sample.id)
```

Comment:
Use `self.stdout.write` or `self.stderr.write` instead of `print` in django-admin commands.

https://docs.djangoproject.com/en/1.9/howto/custom-management-commands/